lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py

```diff
@@ -1172,7 +1172,50 @@ def test_connectivity_names(self):
             )
             self.assertEqual(expected_names, result_names, fail_msg)
 
-    def _check_two_different_meshes(self, vars):
+    def test_multiple_equal_mesh(self):
```

Comment:
NOTE: this bit looks complicated, but only because I removed a subsidiary routine.
Check out the individual commits, which make a lot more sense in this case :
  * https://github.com/SciTools/iris/pull/6004/commits/5365adea44e7b01f4dc55043c62b2cbb1ebe5838
  * https://github.com/SciTools/iris/pull/6004/commits/e57c30d23906525f5b115faec50dbedd31ab37cf